src/main/java/com/example/may/cloudfoundry/destination/service/DestinationService.java

```diff
@@ -0,0 +1,113 @@
+package com.example.may.cloudfoundry.destination.service;
+
+import com.example.may.cloudfoundry.destination.credentials.DestinationCredentials;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException.Unauthorized;
+import org.springframework.web.client.RestTemplate;
+
+import static com.example.may.core.mapper.JsonConverter.stringToModel;
+import static com.example.may.core.ouath.constant.OAuth2Constants.BEARER_AUTHORIZATION_TYPE;
+import static com.example.may.core.ouath.constant.OAuth2Constants.HEADER_NAME_AUTHORIZATION;
+import static java.lang.String.format;
+
+/**
+ * returns properties from destination service.
+ *
+ * @author Evelina Tubalets
+ */
+@Slf4j
+@Service
+@AllArgsConstructor
+public class DestinationService {
+
+    public static final String NAME_OF_DESTINATION_CONFIGURATION_JSON_OBJECT = "destinationConfiguration";
+
+    private final DestinationXsuaaTokenService destinationXsuaaTokenService;
+    private final DestinationCredentials credentials;
+    private final RestTemplate restTemplate;
+
+    /**
+     * uses token for getting properties from destination service.
+     * will retry to call this method in case when token is invalid(expired).
+     *
+     * @return properties from destination service.
+     */
+    @Retryable(value = Unauthorized.class, maxAttempts = 2, backoff = @Backoff(0))
+    public <T> T getProperties(final Class<T> type, final String destinationPath) {
```

Comment:
- rename to getByName()
- destinationPath -> name
- remove type param, you should return generic response:
1. create class Destination with all properties.
2. after that calling side will convert Destination object into required format (in EmailService to EmailEmailProperties in your case)